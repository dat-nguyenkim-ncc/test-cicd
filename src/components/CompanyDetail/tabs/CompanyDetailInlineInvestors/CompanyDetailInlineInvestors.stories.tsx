import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CompanyDetailInlineInvestors, {
  CompanyDetailInlineInvestorsProps,
} from './CompanyDetailInlineInvestors'

export default {
  title: 'Company Details/CompanyDetailInlineInvestors',
  component: CompanyDetailInlineInvestors,
} as Meta

const Template: Story<CompanyDetailInlineInvestorsProps> = args => (
  <CompanyDetailInlineInvestors {...args} />
)

export const Default = Template.bind({})
Default.args = {
  lead: 'Accel Partners, VC',
  other:
    'Goodwater Capital, VC Orange, Corporate Passion Capital, VC Reference Capital (formerly Genevest), VC Stripe, Tech Thrive Capital, VC Vanderbilt University, Other Y Combinator, Accelerator',
} as CompanyDetailInlineInvestorsProps
