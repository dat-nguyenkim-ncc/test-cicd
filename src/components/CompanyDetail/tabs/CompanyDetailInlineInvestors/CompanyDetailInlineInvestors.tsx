import React, { useState } from 'react'
import { Box, Flex } from 'theme-ui'
import { InvestorsProps } from '../../../../types'
import strings from '../../../../strings'
import Icon from '../../../Icon'
import { Paragraph } from '../../../primitives'

export type CompanyDetailInlineInvestorsProps = InvestorsProps & {}

const CompanyDetailInlineInvestors = ({ lead, other }: CompanyDetailInlineInvestorsProps) => {
  const [opened, setOpened] = useState(!lead)
  const {
    companyDetail: { financials: copy },
  } = strings

  const onToggle = () => {
    setOpened(!opened)
  }

  return (
    <Box>
      {lead && (
        <>
          <Flex
            sx={{ cursor: other ? 'pointer' : 'default', alignItems: 'start' }}
            onClick={other ? onToggle : undefined}
          >
            <Paragraph sx={{ mb: 2, mr: 2 }} light>
              {copy.fundingRounds.leadInvestor}
            </Paragraph>
            {other && <Icon size="tiny" icon={opened ? 'indicatorUp' : 'indicatorDown'} />}
          </Flex>
          <Paragraph sx={{ mb: 4 }}>{lead}</Paragraph>
        </>
      )}
      {other && opened && (
        <>
          <Paragraph sx={{ mb: 2 }} light>
            {copy.fundingRounds.otherInvestors}
          </Paragraph>
          <Paragraph>{other}</Paragraph>
        </>
      )}
    </Box>
  )
}

export default CompanyDetailInlineInvestors
