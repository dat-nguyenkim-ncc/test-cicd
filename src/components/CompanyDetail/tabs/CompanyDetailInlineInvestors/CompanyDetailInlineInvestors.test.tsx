import * as React from 'react'
import { shallow } from 'enzyme'
import CompanyDetailInlineInvestors from './CompanyDetailInlineInvestors'

describe('CompanyDetailInlineInvestors', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<CompanyDetailInlineInvestors />)
    expect(wrapper).toMatchSnapshot()
  })
})
