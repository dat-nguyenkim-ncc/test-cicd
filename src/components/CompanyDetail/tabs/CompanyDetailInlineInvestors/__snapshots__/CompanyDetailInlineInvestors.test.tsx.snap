// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetailInlineInvestors matches snapshot 1`] = `ShallowWrapper {}`;
